hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/initialization/BlocklistAccountCreator.java

```diff
@@ -148,10 +152,27 @@ private List<BlockedInfo> parseBlockList(final List<String> fileLines) {
     }
 
     @NonNull
-    private List<String> readPrivateKeyBlocklist(final @NonNull String fileName) {
-        final var inputStream = getClass().getClassLoader().getResourceAsStream(fileName);
-        final var reader = new BufferedReader(new InputStreamReader(inputStream));
-        return reader.lines().toList();
+    private List<String> readPrivateKeyBlocklist(
+            final @NonNull String defaultResource, final @NonNull String resourcePath) {
+        log.info("Bootstrapping blocklist from '{}'", resourcePath);
+        try (var externalResourceStream = Files.newInputStream(Paths.get(resourcePath));
+                var externalResourceReader = new BufferedReader(new InputStreamReader(externalResourceStream))) {
+            return externalResourceReader.lines().toList();
+        } catch (final IOException e) {
+            log.warn("Could not read external properties file {}", resourcePath);
+        }
+
+        log.info("Bootstrapping blocklist from default resource '{}'", defaultResource);
+        try (final var inputStream = getClass().getClassLoader().getResourceAsStream(defaultResource)) {
+            if (inputStream == null) {
+                throw new IOException("Could not find default resource " + defaultResource);
+            }
+            final var reader = new BufferedReader(new InputStreamReader(inputStream));
+            return reader.lines().toList();
```

Comment:
This exception is caught in the same block, which loses the message and effectively uses the exception as a goto.
Possible rewrite:
```suggestion
            if (inputStream != null) {
                final var reader = new BufferedReader(new InputStreamReader(inputStream));
                return reader.lines().toList();
            } else {
                log.warn("Could not find default resource {}.", defaultResource);
                return Collections.emptyList();
            }
```

Reply:
Fixed.